zapcore/write_syncer.go

```diff
@@ -83,6 +84,62 @@ func (w writerWrapper) Sync() error {
 	return nil
 }
 
+type bufferedWriter struct {
+	size int // in bytes
+	buf  []byte
+	ws   WriteSyncer
+}
+
+// Buffer creates a WriteSyncer that buffers the specified number of bytes in
+// memory. Buffered WriteSyncers aren't safe for concurrent use, and should be
+// wrapped by Lock in production.
```

Comment:
nit: the "in production" end sounds weird to me, why not just stop at saying "aren't safe for concurrent use."?  Whether or not something is used concurrently doesn't seem to me to be function of its deployment environment ;-)